test/highlvl-async-usage.test.js

```diff
@@ -172,6 +172,25 @@ describe('basic usage of high-level async methods', function(){
 
   });//</it should have aggregated some config as expected>
 
+  it('should maintain object references when merging modules together', function(done) {
+
+    loader.aggregate({
+      dirname: path.resolve(__dirname, './fixtures/test-refs/config'),
+      filter: /(.+)\.js$/
+    }, function (err, config){
+      if (err) { return done(err); }
+
+      assert(config.databases.someDb);
```

Comment:
(this also might be fixed in mocha v3- I have no idea.  I just haven't spent the time to get to the bottom of it, so I've been trying to avoid any potential problems)